src/main/java/duke/util/Storage.java

```diff
@@ -0,0 +1,114 @@
+    private final String fileDirPath;
+    private final String fullFilePath;
+    private File file;
+
+    /**
+     * Constructor for Storage.
+     *
+     * @param fileName Name of data file.
+     * @param fileDirPath File directory path of data file.
+     * @throws DukeException If there are issues creating or reading the file.
+     */
+    public Storage(String fileName, String fileDirPath) throws DukeException {
+        this.fileDirPath = fileDirPath;
+        fullFilePath = fileDirPath + "/" + fileName;
+
+        initFile();
+    }
+
+    /**
+     * Initializes file and directory.
+     *
+     * <p>Creates file and directory if they don't exist yet base on the file name and directory path provided.
+     * If file and directory already exists, will just open the file.</>
+     *
+     * @throws DukeException If cannot create new file.
+     */
+    private void initFile() throws DukeException {
+        File directory = new File(fileDirPath);
+
+        // create directory if exist
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+        // check if file exist
+        file = new File(fullFilePath);
+        if (!file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException exception) {
+                throw new DukeException(FILE_CANNOT_CREATE_MSG);
+            }
+        }
+    }
+
+    /**
+     * Load task list from file saved.
+     *
+     * @param list List to store the new objects created based on the data.
+     * @param factory Function<String, T> that takes in an input and returns a new object that extends Loading.
+     * @param <T> Object that should extend Loading.
+     * @throws DukeException If there are issues reading the file or format of file is wrong.
+     */
+    public <T extends Loading> void loadFromSave(ArrayList<T> list,
+            Function<String, T> factory)
+            throws DukeException {
+        try {
+            // read file
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String nextLine = scanner.nextLine();
+                T newTask = factory.apply(nextLine); // create the proper obj type
+
+                newTask.extractFileData(nextLine);
+                list.add(newTask);
+            }
+
+            scanner.close();
+
+        } catch (IOException exception) {
+            throw new DukeException(FILE_LOADING_ERROR_MSG);
+        }
+    }
+
+    /**
+     * Save data in list to a file.
+     *
+     * @param list A list that data needs to be saved. The objects must extend Saving.
+     * @throws DukeException If there is problem writing to the file.
+     */
+    public void saveList(ArrayList<? extends Saving> list) throws DukeException {
+        try {
+            // write data to file
+            FileWriter fileWriter = new FileWriter(file);
+            for (Saving item : list) {
```

Comment:
Spacing between item and : is unnecessary.

Reply:
Sorry, I know the code will still compile and run without the space between item and :, but may I know if without space is recommended by the style guide? I can't find any info about it on the java coding standard page and most of the documentation I search through showed a space in between to improve readability.